src/gazebo/differential_drive_6w.cpp

```diff
@@ -195,130 +189,106 @@ namespace gazebo {
+        d1 = stepTime.Double() * mWheelDiameter / 2 * mJoints[MID_LEFT]->GetVelocity(0);
+        d2 = stepTime.Double() * mWheelDiameter / 2 * mJoints[MID_RIGHT]->GetVelocity(0);
 
         dr = (d1 + d2) / 2;
-        da = (d1 - d2) / wheelSep;
+        da = (d1 - d2) / mWheelSeparation;
 
         // Compute odometric pose
-        odomPose[0] += dr * cos(odomPose[2]);
-        odomPose[1] += dr * sin(odomPose[2]);
-        odomPose[2] += da;
+        mOdomPose[0] += dr * std::cos(mOdomPose[2]);
+        mOdomPose[1] += dr * std::sin(mOdomPose[2]);
+        mOdomPose[2] += da;
 
         // Compute odometric instantaneous velocity
-        odomVel[0] = dr / stepTime.Double();
-        odomVel[1] = 0.0;
-        odomVel[2] = da / stepTime.Double();
-
-        if (enableMotors) {
-            joints[FRONT_LEFT]->SetVelocity(0, wheelSpeed[0] / (wheelDiam / 2.0));
-            joints[MID_LEFT]->SetVelocity(0, wheelSpeed[0] / (wheelDiam / 2.0));
-            joints[REAR_LEFT]->SetVelocity(0, wheelSpeed[0] / (wheelDiam / 2.0));
-
-            joints[FRONT_RIGHT]->SetVelocity(0, wheelSpeed[1] / (wheelDiam / 2.0));
-            joints[MID_RIGHT]->SetVelocity(0, wheelSpeed[1] / (wheelDiam / 2.0));
-            joints[REAR_RIGHT]->SetVelocity(0, wheelSpeed[1] / (wheelDiam / 2.0));
-
-            joints[FRONT_LEFT]->SetEffortLimit(0, torque);
-            joints[MID_LEFT]->SetEffortLimit(0, torque);
-            joints[REAR_LEFT]->SetEffortLimit(0, torque);
-
-            joints[FRONT_RIGHT]->SetEffortLimit(0, torque);
-            joints[MID_RIGHT]->SetEffortLimit(0, torque);
-            joints[REAR_RIGHT]->SetEffortLimit(0, torque);
+        mOdomVelocity[0] = dr / stepTime.Double();
+        mOdomVelocity[1] = 0.0;
+        mOdomVelocity[2] = da / stepTime.Double();
+
+        if (mEnableMotors) {
+            mJoints[FRONT_LEFT]->SetVelocity(0, mWheelSpeeds[0] / (mWheelDiameter / 2.0));
+            mJoints[MID_LEFT]->SetVelocity(0, mWheelSpeeds[0] / (mWheelDiameter / 2.0));
+            mJoints[REAR_LEFT]->SetVelocity(0, mWheelSpeeds[0] / (mWheelDiameter / 2.0));
+
+            mJoints[FRONT_RIGHT]->SetVelocity(0, mWheelSpeeds[1] / (mWheelDiameter / 2.0));
+            mJoints[MID_RIGHT]->SetVelocity(0, mWheelSpeeds[1] / (mWheelDiameter / 2.0));
+            mJoints[REAR_RIGHT]->SetVelocity(0, mWheelSpeeds[1] / (mWheelDiameter / 2.0));
+
+            mJoints[FRONT_LEFT]->SetEffortLimit(0, mTorque);
+            mJoints[MID_LEFT]->SetEffortLimit(0, mTorque);
+            mJoints[REAR_LEFT]->SetEffortLimit(0, mTorque);
+
+            mJoints[FRONT_RIGHT]->SetEffortLimit(0, mTorque);
+            mJoints[MID_RIGHT]->SetEffortLimit(0, mTorque);
+            mJoints[REAR_RIGHT]->SetEffortLimit(0, mTorque);
         }
 
         publish_odometry();
     }
 
-    // NEW: Now uses the target velocities from the ROS message, not the Iface
     void DiffDrivePlugin6W::GetPositionCmd() {
-        lock.lock();
+        std::lock_guard guard(mLock);
```

Comment:
can you scope the part that actually needs to be locked with brackets, otherwise its wasteful.

Reply:
Do you care about this too much or can I resolve this.

Reply:
uh i mean it's fine ig

Reply:
All you have to do is add two brackets tho lol, but you can resolve